src/components/CopyToClipboard/CopyToClipboard.tsx

```diff
@@ -0,0 +1,30 @@
+import copyText from 'lib/copyText'
+import * as React from 'react'
+
+type Props = {
+  text: string
+  children: React.ReactNode
+  role: 'button' | 'option'
+  className?: string
+  onCopy?: () => void
+}
+
+//TODO: if this copyText is not passed anywhere else then just remove this param
+export default class CopyToClipboard extends React.PureComponent<Props> {
```

Comment:
Just out of curiosity, are you using a HOC because there are components that don't have an `onClick` prop?

Reply:
So instead of HOC should I implement this with just `copyText` function as it  is Marketplace?